src/controllers/users.js

```diff
@@ -0,0 +1,114 @@
+import bcrypt, { compareSync } from 'bcrypt';
+import user from '../db/models';
+import { generateToken } from '../middlewares/authentication';
+
+const { User } = user;
+
+/**
+  * @description class representing User Authentication
+  *
+  * @class UserHandler
+  */
+class userHandler {
+  /**
+    * @description - This method is responsible for creating new users
+    *
+    * @static
+    * @param {object} request - Request sent to the router
+    * @param {object} response - Response sent from the controller
+    *
+    * @returns {object} - object representing response message
+    *
+    * @memberof UserHandler
+    */
+  static async registerUser(request, response) {
+    const { username, email } = request.body;
+    const password = bcrypt.hashSync(request.body.password, 10);
```

Comment:
Any specific reason why `hashSync` (blocking) was preferred over `hash` (non-blocking) here? (per [this notice](https://www.npmjs.com/package/bcrypt#why-is-async-mode-recommended-over-sync-mode))

Reply:
The app is not a robust one, so I didn't see a difference between Asynchronous and Synchronous version here. In a robust application, the Async version is preferred because the hashing process can be CPU intensive and Sync version may block an event loop.
Thanks for taking time to peruse the codebase thoroughly.